LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/CommonAPI/ApiVersion.cs

```diff
@@ -254,29 +253,17 @@ public int CompareTo(ApiVersion other)
             Name == version.Name &&
             IsKnown == version.IsKnown;
 
-        public static bool operator <(ApiVersion value1, ApiVersion value2)
-        {
-            if (value1 is null) throw new ArgumentNullException(nameof(value1));
-            return value1.CompareTo(value2) < 0;
-        }
+        public static bool operator <(ApiVersion value1, ApiVersion value2) =>
```

Comment:
I think the whole comparison stuff might not be fully tested or even flawed so why not just remove it instead of adding more fixes and tests. It doesn't seem to be used anywhere except in `SupportsVersion`, where it can be all in-lined. This could be done in a separate PR that's merged into this.

Reply:
I added tests for the comparison before addressing the warnings - did I miss a case there? Curious to learn, why should it be flawed, is it because of comparing to `null`?

Reply:
> Curious to learn, why should it be flawed…?

Because it seems wrong to me that the equality implementation considers the 3 properties of a version (`Version`, `Name` and `IsKnown`) yet comparison only cares about `Version`, and what's more, `CompareTo` initially depends on equality/`Equals`. This tells me there's some indecision about what fundamentally constitutes as the identity or value of the type and this can lead to surprises or subtle bugs.

Reply:
Addressed in ebe9bde78806fbf0c0384f25763d9cbbfcdd14a4